### ISSUE TITLE
False positive on object methods with arrow functions inside

### ISSUE THREAD

**Issue by ChALkeR** [inspect-js/is-arrow-function#26]
```js
const isArrow = require('is-arrow-function')
const x = {
  foo() {
    return 42
  },
  bar() {
    return (() => 10)()
  }
}
console.log(x.foo(), x.bar())
console.log(isArrow(x.foo))
console.log(isArrow(x.bar)) // fail
```

**Comment by ljharb** [comment]
This feels similar to #15 - short of including an actual JS parser, I'm not sure how to reliably handle cases like these.
